### PR TITLE
Generic Target Mode

### PR DESCRIPTION
**Pull request by thheinen**
First draft PR of Target Mode to check compatibility with pipelines

### CODE REVIEW

**Comment by tpowell-progress** [comment]
@thheinen is a rebase on `main` possible from the state of your branch?

File: lib/chef/provider/user.rb [hunk 1]
```diff
@@ -153,11 +156,7 @@ def compare_user
           new_val = new_resource.send(user_attrib)
           cur_val = current_resource.send(user_attrib)
           if !new_val.nil? && new_val.to_s != cur_val.to_s
-            if user_attrib.to_s == "password" && new_resource.sensitive
-              @change_desc << "change #{user_attrib} from ******** to ********"
-            else
-              @change_desc << "change #{user_attrib} from #{cur_val} to #{new_val}"
-            end
```
Comment (neha-p6) [inline lib/chef/provider/user.rb:160]: This block needs to be kept as it is. It is to mask password property when it is set as sensitive.

Comment (thheinen) [reply lib/chef/provider/user.rb]: Hi, yeah it was some regression on my side which I asked John about. Fixed in the following commit - thanks for spotting!

**Comment by neha-p6** [comment]
@thheinen If you could rebase to main, that will fix bunch of failures in the verify pipeline. We have merged the `cheffish` gem but the test can move ahead once rebase pulled in current fixes.

**Review by neha-p6 — APPROVED** [review]

**Comment by neha-p6** [comment]
Pending failures which we will address later on https://buildkite.com/chef-oss/chef-chef-main-verify/builds/17905#_

File: lib/chef/provider/package/apt.rb [hunk 2]
```diff
@@ -124,11 +124,6 @@ def unlock_package(name, version)
 
         private
 
-        # @return [String] package name with or without anchors attached to it.
-        def resolve_package(pkg)
-          new_resource.anchor_package_regex ? "^#{pkg}$" : pkg
-        end
-
```
Comment (neha-p6) [inline lib/chef/provider/package/apt.rb:131]: @thheinen This looks like a regression. This was done to fix a bug fyi https://github.com/chef/chef/pull/13873

**Pull request merged**